File: app/src/main/java/intellij/class13/deliverableexercise/list/java/Teacher.java
package intellij.class13.deliverableexercise.list.java;

import java.util.Objects;

public abstract class Teacher {
    private String name;
    private String lastName;
    private Integer houseTime;
    private Integer teacherCode;

    public Teacher(String name, String lastName, Integer houseTime, Integer teacherCode) {
        this.name = name;
        this.lastName = lastName;
        this.houseTime = houseTime;
        this.teacherCode = teacherCode;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getLastName() {
        return lastName;
    }

    public void setLastName(String lastName) {
        this.lastName = lastName;
    }

    public Integer getHouseTime() {
        return houseTime;
    }

    public void setHouseTime(Integer houseTime) {
        this.houseTime = houseTime;
    }

    public Integer getTeacherCode() {
        return teacherCode;
    }

    public void setTeacherCode(Integer teacherCode) {
        this.teacherCode = teacherCode;
    }

    //Equals padrão que compara o objeto inteiro
//    @Override
//    public boolean equals(Object o) {
//        if (this == o) return true;
//        if (o == null || getClass() != o.getClass()) return false;
//        Teacher teacher = (Teacher) o;
//        return Objects.equals(name, teacher.name) &&
//                Objects.equals(teacherCode, teacher.teacherCode);
//    }

    //Equals que compara apenas o Código do Professor do objeto
    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        Teacher teacher = (Teacher) o;
        return Objects.equals(teacherCode, teacher.teacherCode);
    }

    @Override
    public String toString() {
        return "'" + name + " " + lastName + '\'';
    }
}

File: app/src/main/java/intellij/class13/deliverableexercise/list/kotlin/Teacher.kt
package intellij.class13.deliverableexercise.list.kotlin

abstract class Teacher(var name: String, var lastName: String, var houseTime: Int, var teacherCode: Int) {

    //Equals padrão que compara o objeto inteiro
//    override fun equals(other: Any?): Boolean {
//        if (this === other) return true
//        if (javaClass != other?.javaClass) return false
//        other as Teacher
//        if (name != other.name) return false
//        if (lastName != other.lastName) return false
//        if (houseTime != other.houseTime) return false
//        if (teacherCode != other.teacherCode) return false
//        return true
//    }

    //Equals que compara apenas o Código do Professor do objeto
    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other == null || javaClass != other.javaClass) return false
        val teacher = other as Teacher
        return teacherCode == teacher.teacherCode
    }

    override fun toString(): String {
        return "'$name $lastName'"
    }
}